### PR TITLE
util: add basic ini parser

### PR DESCRIPTION
**Pull request by jbedard**
Is this about what we're looking for? A basic skylark method that can be used anywhere... 🤷 

idk if we have a standard for where to put private utils like this?

### CODE REVIEW

**Comment by alexeagle** [comment]
Is there anything JS-specific about this? I would have imagined it belongs in bazel-lib so that we could use it in all our rulesets.

(Note, the yaml parser is JS-specific because we only bothered to implement the subset that pnpm-lock files cover)

**Comment by jbedard** [comment]
This is also only a subset which npmrc requires, but I agree it should be in bazel-lib if/when it is a full parser. See the comment specifying what is not supported.

**Review by alexeagle — APPROVED** [review]
move to `npm/private` next to yaml.bzl then, and just call `ini.bzl`

**Comment by jbedard** [comment]
Note we'll also have a dumb partial glob implementation. I think that could also go into `npm/private`?

File: npm/private/test/BUILD.bazel [hunk 1]
```diff
@@ -4,13 +4,15 @@ load(":transitive_closure_tests.bzl", "transitive_closure_tests")
...
 load(":utils_tests.bzl", "utils_tests")
+load(":ini_test.bzl", "init_tests")
 
 # gazelle:exclude *_checked.bzl
 
 npm_link_all_packages(name = "node_modules")
 
 # Unit tests
 utils_tests(name = "test_utils")
+init_tests(name = "test_ini")
```
Comment (alexeagle) [inline npm/private/test/BUILD.bazel:15]: init rather than ini ?

Comment (jbedard) [reply npm/private/test/BUILD.bazel:15]: I keep doing that 😡

Thank you lol

**Review by alexeagle — APPROVED** [review]

**Pull request merged**